src/db/parity_db.rs

```diff
@@ -140,16 +190,36 @@ impl Blockstore for ParityDb {
         D: AsRef<[u8]>,
         I: IntoIterator<Item = (Cid, D)>,
     {
-        let values = blocks
+        let values = blocks.into_iter().map(|(k, v)| {
+            let column = Self::choose_column(&k);
+            match column {
+                DbColumn::GraphDagCborBlake2b256 | DbColumn::Settings | DbColumn::GraphFull => {
+                    (column, k.to_bytes(), v.as_ref().to_vec())
+                }
+            }
+        });
+        let tx = values
             .into_iter()
-            .map(|(k, v)| (k.to_bytes(), v.as_ref().to_vec()));
-        self.bulk_write(values).map_err(|e| e.into())
+            .map(|(col, k, v)| (col as u8, Operation::Set(k, v)));
+        self.db
+            .commit_changes(tx)
+            .map_err(|e| anyhow!("error bulk writing: {e}"))
     }
 }
 
 impl BitswapStoreRead for ParityDb {
     fn contains(&self, cid: &Cid) -> anyhow::Result<bool> {
-        Ok(self.exists(cid.to_bytes())?)
+        for column in [DbColumn::GraphDagCborBlake2b256, DbColumn::GraphFull] {
```

Comment:
Interesting optimization. Maybe add a note saying that this is faster because 99.99% of all values are in the `GraphDagCborBlake2b256` column. If that assumption changes in the future, this code will become slower.